src/state_machine.zig

```diff
@@ -835,14 +843,14 @@ pub fn StateMachineType(
                         return;
                     }
                 }
-            } else {
-                // TODO(batiati): Improve the way we do validations on the state machine.
-                log.info(
-                    "invalid filter for get_account_balances: {any}",
-                    .{filter},
-                );
             }
 
+            // TODO(batiati): Improve the way we do validations on the state machine.
+            log.info(
+                "invalid filter for get_account_balances: {any}",
```

Comment:
I think this error message is not necessarily correct anymore. Unless querying an account that doesn't exist or doesn't set `history` qualifies as an "invalid filter"? But we can fix this later, seeing as this is just placeholder error handling anyhow.